aif360/datasets/structured_dataset.py

```diff
@@ -149,6 +149,32 @@ class description.
             privileged_protected_attributes=privileged_protected_attributes,
             metadata=metadata)
 
+        
+    def subset(self, indexes):
+        """ Subset of dataset based on instance names
+        Args:
+            indexes: iterable which contains instance names 
+
+        Returns:
+            `StructuredDataset`: subset of dataset based on indexes
+        """
+
+        # convert each element of indexes to string
+        indexes_str = [str(x) for x in indexes]
+        # check if indexes is a subset of instance_names
+        if not set(indexes_str).issubset(set(self.instance_names)):
+            raise ValueError("indexes must be a subset of instance_names.")
```

Comment:
This isn't actually necessary since the instance names aren't required to just be string-versions of integer values. They could, for example, be names of people. I think integer-indexing should be sufficient. Otherwise, looks good!

Reply:
I also think that integer indexing should be good enough but I had to take this extra route because of the way instance_names have been defined in __init__ function. 

Please refer to structured_dataset.py  line 112
`
self.instance_names = df.index.astype(str).tolist()
`

Without this extra code, instance_names won't be consistent between the original dataset and its subset. instances_names for original dataset will be a list of strings while for its subset it will be list of integers.

Furthermore, this extra code also helps check for inconsistent index values like negative integers.   

Correction: this subset function is based on position and not on row names.

Reply:
Please check the latest changes. Now, subset() works on row names rather than position.

For example, it can deal with the following case
'
ad = AdultDataset()
ad.instance_names[0] = 'i0'
ad.instance_names[4] = 'i4'
ad.subset(['i0', 'i4'])
'